### PR TITLE
Fix post-uninstall warnings

### PR DESCRIPTION
**Pull request by ArthurKamalov**
This PR, to reduce noise, removes some warnings that are produced during the package uninstallation. 

Other changes: 
* It also disables check of the new version of the pip, so the update suggestion message won't shown every time.

* Also corrected venv removal during the `pre-uninstall` script execution.

* Retries added to the package status tests to reduce overall flakiness.

### CODE REVIEW

File: agent_build_refactored/managed_packages/files/bin/agent-libs-config [hunk 1]
```diff
@@ -52,6 +52,7 @@ def initialize():
             "pip",
             "install",
             "--quiet",
+            "--disable-pip-version-check",
```
Comment (tomaz-s1) [inline agent_build_refactored/managed_packages/files/bin/agent-libs-config:55]: 👍 

Probably not a bad idea for us to have some check in CI/CD which periodically reminds us to upgrade pip in the venv or similar.

File: tests/end_to_end_tests/tools.py [hunk 2]
```diff
@@ -89,6 +89,28 @@ def start(self, no_fork: bool = False, env=None):
...
+        time.sleep(0.2)
+
+        attempts = 0
+
+        while True:
+            try:
+                self.get_status_json()
+            except subprocess.CalledProcessError as e:
+                if attempts >= 10:
+                    raise e
```
[the remark this answers is not in the document]
Comment (ArthurKamalov) [reply tests/end_to_end_tests/tools.py:113]: Sure, added more log messages, but also increased initial wait so it has more time before it throws its first warning.

**Review by tomaz-s1 — APPROVED** [review]
LGTM 👍

File: agent_build_refactored/managed_packages/files/bin/agent-libs-config (hunk 1 above)
Comment (ArthurKamalov) [reply agent_build_refactored/managed_packages/files/bin/agent-libs-config:56]: yeah, especially with the fact that we cache our builds.

**Pull request merged**